### DISCUSSION TITLE
Saving the latent representation

### DISCUSSION THREAD

**Discussion by jonaballe** [tensorflow/compression#78] [General]
Hi @ankitlat,

> I am eager to know is there a way to save encoded latent representations of the pre-trained models? Presently I can only save the compressed byte stream.

Yes, you can use the "dump" subcommand of tfci.py to save any of the internal tensors of the model to NumPy format. However, you'll need to know the name of the tensor you want.

For example, in the b2018-gdn-128-1 model, you presumably want "analysis/layer_2/convolution:0".

Hope this helps!
Johannes

Originally posted by @ankitlat in #76 (comment)

**Comment by ankitlat** [comment]
> Hi @ankitlat,
> 
> > I am eager to know is there a way to save encoded latent representations of the pre-trained models? Presently I can only save the compressed byte stream.
> 
> Yes, you can use the "dump" subcommand of tfci.py to save any of the internal tensors of the model to NumPy format. However, you'll need to know the name of the tensor you want.
> 
> For example, in the b2018-gdn-128-1 model, you presumably want "analysis/layer_2/convolution:0".
> 
> Hope this helps!
> Johannes
> 
> Originally posted by @ankitlat in #76 (comment)

Thanks a lot!

**Comment by Saleh-Gholam-Zadeh** [comment]
Thank you @jonycgn,

I have the same question, however I am working with the  bmshj2018-hyperprior-msssim-1 model and I want to obtain the correct name of the tensor which represents the latent space 
How can I have access to it

Thank you in advance
